assembly/decoder.ts

```diff
@@ -76,8 +74,8 @@ let CHAR_A_LOWER = "a".charCodeAt(0);
 
 export class DecoderState {
     readIndex: i32 = 0;
-    buffer: Uint8Array = null;
-    lastKey: string = null;
+    buffer: Uint8Array;
```

Comment:
hmm, why `buffer` isn't initialized to `null` anymore?

Reply:
Using the tsconfig from assemblyscript's repo, my IDE complained that buffer could not be assigned to null since it wasn't an explicitly in the type.  However, adding it to the type means that everywhere it's used it could be null and either you have to check or use `buffer!`.  I tested it and it generates the same binary with and without this null.  When allocating a new `DecoderState` it will just allocated the space for the reference.

Since a `DecoderState` should always have a buffer defined, I just changed `DecoderState` to have a constructor that makes the buffer mandatory and the other options have their current defaults.